src/App.tsx

```diff
@@ -15,49 +45,190 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const goodsWithId: GoodWithID[] = goodsFromServer.map(good => (
+  {
+    name: good,
+    // eslint-disable-next-line
+    id: self.crypto.randomUUID(),
+  }
+));
+
+const SORT_VALUE_NAME = 'name';
+const SORT_VALUE_LENGTH = 'length';
+const SORT_VALUE_REVERSE = 'reverse';
+const SORT_VALUE_REREVERSE = 'reReverse';
+const SORT_VALUE_RESET = 'reset';
+
+const activeFields: ActiveFields = {
+  name: false,
+  length: false,
+  reverse: false,
+  reReverse: false,
+  reset: false,
+  initial: true,
+};
+
+let canChangeState = false;
+let previousActivField = '';
+
+function getPreperedGoods(goods: GoodWithID[], sortField: keyof ActiveFields) {
+  previousActivField = sortField;
+
+  // eslint-disable-next-line
+  switch (true) {
+    case sortField === SORT_VALUE_REREVERSE:
+      activeFields.reverse = !activeFields.reverse;
+      break;
+
+    case sortField === SORT_VALUE_NAME && !activeFields.name:
+      activeFields.length = false;
+      canChangeState = true;
+      break;
+
+    case sortField === SORT_VALUE_LENGTH && !activeFields.length:
+      activeFields.name = false;
+      canChangeState = true;
+      break;
```

Comment:
try to simplify all this logic with getPreperedGoods and activeFields, you can only check if `sortField === SORT_VALUE_NAME` and `sortField === SORT_VALUE_LENGTH`